### PR TITLE
Doc hint fs tools

### LINKED ISSUE #1691: XFS build fails - grub2-i386-pc
## Problem description

I try to build SLES 15 SP 2 with root filesystem on xfs. Building with ext4 works fine. Building with xfs fails.

```
INFO: 08:25:20 | Installing grub2 on disk /dev/loop0
DEBUG: 08:25:20 | EXEC: [mountpoint -q /tmp/kiwi_mount_manager.r4j126ag]
DEBUG: 08:25:20 | EXEC: [mount /dev/mapper/loop0p3 /tmp/kiwi_mount_manager.r4j126ag]
DEBUG: 08:25:20 | EXEC: [mountpoint -q /tmp/kiwi_mount_manager.r4j126ag/boot/efi]
DEBUG: 08:25:20 | EXEC: [mount /dev/mapper/loop0p2 /tmp/kiwi_mount_manager.r4j126ag/boot/efi]
DEBUG: 08:25:20 | EXEC: [mountpoint -q /tmp/kiwi_mount_manager.r4j126ag/dev]
DEBUG: 08:25:20 | EXEC: [mount -n --bind /dev /tmp/kiwi_mount_manager.r4j126ag/dev]
DEBUG: 08:25:20 | EXEC: [mountpoint -q /tmp/kiwi_mount_manager.r4j126ag/proc]
DEBUG: 08:25:20 | EXEC: [mount -n --bind /proc /tmp/kiwi_mount_manager.r4j126ag/proc]
DEBUG: 08:25:20 | EXEC: [mountpoint -q /tmp/kiwi_mount_manager.r4j126ag/sys]
DEBUG: 08:25:20 | EXEC: [mount -n --bind /sys /tmp/kiwi_mount_manager.r4j126ag/sys]
DEBUG: 08:25:20 | EXEC: [chroot /tmp/kiwi_mount_manager.r4j126ag grub2-install --skip-fs-probe --directory /usr/lib/grub2/i386-pc --boot-directory /boot --target i386-pc --modules ext2 iso9660 linux echo configfile search_label search_fs_file search search_fs_uuid ls normal gzio png fat gettext font minicmd gfxterm gfxmenu all_video xfs btrfs lvm luks gcry_rijndael gcry_sha256 gcry_sha512 crypto cryptodisk test true loadenv multiboot part_gpt part_msdos biosdisk vga vbe chain boot /dev/loop0]
DEBUG: 08:25:21 | EXEC: Failed with stderr: Installing for i386-pc platform.
grub2-install: error: unknown filesystem.
, stdout: (no output on stdout)
ERROR: 08:25:21 | KiwiCommandError: chroot: stderr: Installing for i386-pc platform.
grub2-install: error: unknown filesystem.
, stdout: (no output on stdout)
INFO: 08:25:21 | Cleaning up BootLoaderInstallGrub2 instance
```

## Expected behaviour

Successfully building SLES 15 SP 2.

## Steps to reproduce the behaviour

1. Schema = 7.2 (older schemas also failing)

`<image schemaversion="7.2" name="test-xfs-sles-15">`

2. Type / Disk

```
<type 
            image="oem" 
            filesystem="xfs" 
            format="vmdk" 
            firmware="uefi" 
            bootpartition="false" 
            devicepersistency="by-uuid">
            <bootloader name="grub2" timeout="10" />
            <size unit="G">75</size>
            <oemconfig>
                <oem-resize>false</oem-resize>
            </oemconfig>      
</type>
```

3. Packages 
```
<packages type="image" patternType="plusRecommended">
	...
	<!-- XFS -->
        <package name="xfsprogs"/>
	...
</packages >
```

## OS and Software information
+ KIWI version: KIWI (next generation) version 9.23.5
+ Operating system: openSUSE Tumbleweed
+ OBS version: -

### PR DESCRIPTION
**Pull request by schaefi**
**Added new Troubleshooting chapter to the docs**
    
Added a new chapter to handle issues which depends on several environment conditions and for which not only one possible solution exists. This Fixes #1691

**Deleted legacy kiwi chapter from the docs**
    
The information there is almost obsolete since kiwi(perl) does no longer exist in any modern distribution. It's only maintained on SLE12 which itself is EOL and the kiwi there has its own documentation which we don't need to reference anymore in the upstream project

### CODE REVIEW

**Review by davidcassany — APPROVED** [review]
Looks good to me :+1:

I am wondering if `https://osinside.github.io/kiwi/image_types_and_results.html#build-host-constraints` would not be a nice start initial warning for the troubleshooting section. We could also point to the boxed plugin as a solution to workaround to `host vs image distros` together with OBS option.

**Comment by schaefi** [comment]
> Looks good to me 
> 
> I am wondering if [build-host-constraints](https://osinside.github.io/kiwi/image_types_and_results.html#build-host-constraints) would not be a nice start initial warning for the troubleshooting section. We could also point to the boxed plugin as a solution to workaround to `host vs image distros` together with OBS option.

good idea, I did that. I did not quite understand the obs option you mentioned ?

File: doc/source/troubleshooting.rst [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+For building images a host system is required that runs the build process.
+Tools to create the image are used from that host and this creates an
+indirect dependency to the target image. For example; Building an
+Ubuntu image requires the apt and dpkg tools and metadata to be available
+and functional on the host to build an Ubuntu image. There are many more
+of those host vs. image dependencies and not all of them can be resolved
+in a clear and clean way.
+
+The most compatible environment is provided if the build host is of the same
+distribution than the target image. This always applies for the Open Build
```
Comment (davidcassany) [inline doc/source/troubleshooting.rst:30]: I was thinking that we could also mention the boxed plugin here, as this is an easy way to match host and target distro.

**Review by davidcassany — COMMENTED** [review]
>  I did not quite understand the obs option you mentioned ?

Yes sorry I was not very explicit. See the comment. Just any idea, not so sure it is easy to fit boxed plugin in there.

File: doc/source/troubleshooting.rst (hunk 1 above)
Comment (schaefi) [reply doc/source/troubleshooting.rst:30]: got it and updated the docs

I think it's a good read now and easy to find

**Pull request merged**